receiver/couchdbreceiver/client.go

```diff
@@ -0,0 +1,139 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package couchdbreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/couchdbreceiver"
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+)
+
+const nodeNamesPath = "/_membership"
+
+// CouchDBClient defines the basic HTTP client interface.
+type CouchDBClient interface {
+	Get(path string) ([]byte, error)
+	GetNodeNames() ([]string, error)
+	GetStats(nodeName string) (map[string]interface{}, error)
+}
+
+var _ CouchDBClient = (*couchDBClient)(nil)
+
+type couchDBClient struct {
+	client *http.Client
+	cfg    *Config
+	logger *zap.Logger
+}
+
+// NewCouchDBClient creates a new client to make requests for the CouchDB receiver.
+func NewCouchDBClient(cfg *Config, host component.Host, logger *zap.Logger) (CouchDBClient, error) {
```

Comment:
Same, this is only going to be used by this package, right?

Reply:
Same goes for all the functions in this file actually. Exporting without a need is basically asking others to import and depend on your code. You probably don't want the added responsibility 😄

Reply:
fixed

Reply:
Still most of the functions in this file are exported and don't need to be.